### DISCUSSION TITLE
Best way to read and track objects from data/ in user-defined R package?

### DISCUSSION THREAD

**Discussion by luke-a-rogers** [ropensci/targets#598] [General]
Hello,

I want to use an object from `data/` as the first step in a `targets` pipeline that is part of a user-defined R package. Ideally the pipeline would read the object, watch it for changes, and track it as a dependency for downstream objects.

I am currently using this code chunk in my `_targets.R` script, and it is working:

```r
list(
  tar_target(watch_new_data, "data/new_data.rda", format = "file"),
  tar_target(use_new_data, read_from_path(watch_new_data))
)
```
where `read_from_path()` is a function that takes a file path as its argument, and returns the associated object:

```r
read_from_path <- function (path) {
  envir <- environment()
  data_name <- load(path, envir = envir)
  get(data_name)
}
```

My question: is there an accepted (or more natural) way to achieve this 'read, watch, and track dependency'  behaviour in the `targets` pipeline? Specifically, for objects stored in `data/` as `.rda` files.

Thanks!
Luke

**Comment by wlandau** [comment]
Your approach looks great to me. I don't think either of us have heard much about this use case because the specific part about `data/` does not tend to come up a lot. The `targets` data store is built for much larger and faster data than can fit comfortably into a package (restricted to < 5 MB on CRAN).

For onlookers, I will just note your complementary thread at https://github.com/ropensci/targets/discussions/588, which covers `usethis::use_data()` as well as https://books.ropensci.org/targets/practices.html#packages-based-invalidation.

**Reply by luke-a-rogers** [reply]
Ahhh, that makes sense about the CRAN data restriction. Thanks!

Luke
